nscplugin/src/main/scala-3/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -122,12 +122,20 @@ trait NirGenExpr(using Context) {
 
       desugarTree(lhsp) match {
         case sel @ Select(qualp, _) =>
-          val rhs = genExpr(rhsp)
+          def rhs = genExpr(rhsp)
```

Comment:
Are you sure `def` is required here? It seems the `rhsp` is evaluated so the two uses of `rhs` don't seem like they would change unless `rhsp` is also a function.

Reply:
Yes, it is needed, otherwise, we might generate a call to `extern` (evaluated as `throw UndefinedError`), which result would not be assigned a an argument for other calls. Also since it's throwing an exception it would not be dead code eliminated - or rather everything after throwing an exception would be treated as dead-code. That's why we need to delay generating `rhsp`. Because `rhs` is always used only once the usage of `def` is better then adding `lazy val` due to overhead, and we don't need to duplicate `genExpr` in case if we would like to stick with `val`

Reply:
Got it. You can see how when reading the code, that using `def` there would generate a question as to why.